compiler/nimeval.nim

```diff
@@ -99,14 +99,16 @@ proc findNimStdLibCompileTime*(): string =
 
 proc createInterpreter*(scriptName: string;
                         searchPaths: openArray[string];
-                        flags: TSandboxFlags = {}): Interpreter =
+                        flags: TSandboxFlags = {},
+                        defines = @[("nimscript", "true")],
```

Comment:
Huh? "true"? Only pass the defined symbols instead.

Reply:
But what if you want to do the equivalent of `-d:someflag:100`?

Reply:
That doesn't work with your approach either iirc, the "true" is caused by a silly implementation.

Reply:
Just tested and it works fine. I added a `const testDefine* {.intdefine.} = 100` in my script, then called `createInterpreter(scriptname, libpaths, defines = @[("nimscript", "true"), ("testDefine", "42")])` and then in my script I `echo`'d out the value and it came out as 42.

Reply:
Ok.